### DISCUSSION TITLE
[Question] Pagination and meta data

### DISCUSSION THREAD

**Discussion by themaiby** [tailflow/laravel-orion-ts#49] [Ideas]
Hello.
How can I receive meta information for pagination like current_page, last_page after model.$query.get() request?

This information stored in every element of array but it seems to me that this is not the right solution.
Thanks.

**Comment by alexzarbn** [comment]
Hi @themaiby,

Currently, you can access an entire response object on each model instance returned from the API.
However, like you mentioned, it is not the best solution for index and search operations.

In the coming versions it might be worth introducing a wrapper class that would store a list of returned models + meta information, something like this:

```typescript
class PaginatedCollection {
  public meta: Meta;
  public items: Array<Model>;
}
```

**Comment by gogl92** [comment]
Hi @themaiby was this ever implemented? how would that look like? I may be able to create a PR

I was thinking of something like:

```
const users = await User.$query().index();
```

and users will have 

```
{
  meta: {
    total: 123,
    per_page: 10,
    current_page: 1,
    prev_page_url: "...",
    next_page_url: "..."
  },
  items: [models...]
}
```

**Comment by hamdallah90** [comment]
Hello,

Thank you for this package, But we need this feature ASAP, its very important
